sake

```diff
@@ -9,7 +9,7 @@ Executes a SilverStripe command"
 	exit 1
 fi
 
-if ! [ -x "$(command -v which)" ]; then
+if ! command -v which >/dev/null 2>&1; then
```

Comment:
```suggestion
command -v which >/dev/null 2>&1
if [ $? -ne 0 ]; then
```
Unless there are reasons against doing it this way, please do it like this per my previous request. 

The test syntax is consistent with other conditions in this file and is an expected syntax for conditions.
Explicitly testing the response code makes it obvious that that's all we actually are interested in for this test.

Reply:
Wow, that was a fast response! :hushed: - I was typing my reasoning, but you already answered. 

I interpreted your problem as follow:

1. PHP and Bash  handle '&&' piping differently and therefore may be confusing to PHP developers
2. The fact that the line exceeded 100 characters.

The reason behind my adaptation to your suggestion was the following:

- `$? -ne 0` is some syntax I only know of Bash scripting and therefore may also be very unfamiliar to a PHP developer.
- With the combination of the two lines I moved them clearer into one context: "If some command doesn't work, then…" (in contrast to your code of: "If a magic variable has something to do with "0"…)

If you don't agree with that reasoning, I'm fine to use your suggestion instead.

Reply:
I think it's totally fine to use bash constructs in a bash context ;p

But I also think that testing a specific thing makes it easier to look up what that thing is. I.e, testing `$? -ne 0` makes it easy to look up what that specific syntax does (i.e. it's testing if the exit code of the previously executed command is not equal to 0).

If the full condition is `! command -v which >/dev/null 2>&1` then you need to know what *all* of that does and will return in order to know that what is being tested is just the exit code. 

For that reason I'd prefer the separated syntax in my suggestion

Reply:
Basically, at some point someone with limited bash experience will see this and be confused regardless of which syntax we use - so by breaking it up into small chunks it should be easier to find out what's happening, imo

Reply:
Yeah, it's a difficult topic since everyone has different experiences, know-how, and approaches coding differently.

Therefore, on such philosophical topics, I'm always siding with the team of that particular project.

Reply:
We're talking about a line of code that checks the availability of `which` for days now.

I mean how many users will execute `sake` without having `which` installed? Zero? :slightly_smiling_face:


BTW I've only used the `type -P` approach to let the code be as readable as it was before, even for non-bash developers:
```sh
if ! [ -x "$(type -P which)" ]; then
  echo "Error: sake requires the 'which' command to operate." >&2
  exit 1
fi
```
The `command -v` is also my personal preferred way. 👍 The @GuySartorelli's code looks easier to me while @NoLdman's code is more elegant.